manifests/init.pp

```diff
@@ -3,30 +3,274 @@
+  if $admin_group {
+    validate_string($admin_group)
+  }
+  if $admin_users {
+    validate_string($admin_users)
+  }
+  if $ro_users {
+    validate_string($ro_users)
+  }
+  if $ro_group {
+    validate_string($ro_group)
+  }
+  validate_integer($check_timeout)
+  #this should probably be a bool
+  validate_integer($debug)
+  validate_re($admin_email, $email_regex)
+  validate_re($admin_pager, $email_regex)
+  #this should probably be a bool
+  validate_integer($stalking)
+  #this should probably be a bool
+  validate_integer($flap_detection)
+  validate_bool($enable_ido)
+  validate_re($ido_db_server, $db_options)
+  validate_re($ido_db_host, $host_regex)
+  validate_integer($ido_db_port)
+  validate_string($ido_db_name)
+  validate_string($ido_db_user)
+  validate_string($ido_db_pass)
+  validate_re($web_db_server, $db_options)
+  validate_re($web_db_host, $host_regex)
+  validate_integer($web_db_port)
+  validate_string($web_db_name)
+  validate_string($web_db_user)
+  validate_string($web_db_pass)
+  validate_re($web_auth_type, $auth_type_options)
+  validate_string($web_auth_name)
+  validate_absolute_path($web_auth_user_file)
+  validate_absolute_path($web_auth_group_file)
+  validate_hash($web_auth_users)
+  validate_re($ldap_security, $ldap_security_options)
+  validate_re($ldap_server, $host_regex)
+  validate_string($ldap_firstname)
+  validate_string($ldap_lastname)
+  validate_string($ldap_email)
+  if $ldap_basedn {
+    validate_string($ldap_basedn)
+  }
+  if $ldap_groupdn {
+    validate_string($ldap_groupdn)
+  }
+  if $ldap_binddn {
+    validate_string($ldap_binddn)
+  }
+  if $ldap_bindpw {
+    validate_string($ldap_bindpw)
+  }
+  validate_string($ldap_userattr)
+  validate_string($ldap_groupattr)
+  if $ldap_filter_extra {
+    validate_string($ldap_filter_extra)
+  }
+  if $ldap_auth_group {
+    validate_string($ldap_auth_group)
+  }
+  validate_absolute_path($nagios_plugins)
+  if $nagios_extra_plugins {
+    validate_absolute_path($nagios_extra_plugins)
+  }
+  validate_string($icinga_cmd_grp)
+  if $db_password {
+    validate_string($db_password)
+  }
+  if $email_user {
+    validate_string($email_user)
+  }
+  if $email_password {
+    validate_string($email_password)
+  }
+  if $ssl_cert_source {
+    validate_absolute_path($ssl_cert_source)
+  }
+  if $clickatell_api_id {
+    validate_string($clickatell_api_id)
+  }
+  if $clickatell_username {
+    validate_string($clickatell_username)
+  }
+  if $clickatell_password {
+    validate_string($clickatell_password)
+  }
+  validate_bool($is_pbx)
+  if $pbx_mngr_pw {
+    validate_($pbx_mngr_pw)
+  }
+  validate_absolute_path($cgi_path)
+  validate_absolute_path($html_path)
+  if $css_path {
+    validate_absolute_path($css_path)
+  }
+  validate_absolute_path($pnp4nagios_html_path)
+  validate_absolute_path($cgi_url)
+
+  if $manage_users {
+    include icinga::users
+    Class['icinga::users'] -> Class['icinga::install']
+  }
+  if $manage_dbs {
```

Comment:
I tend to leave database management out of my modules and leave it as something people can manage with their profiles.
Everyone does it different (I use mariadb and galera with exported resources) so it's not worth the trouble of attempting to support one use case.